### ISSUE TITLE
Status of NumPy compatibility

### ISSUE THREAD

**Issue by rgommers** [Quansight-Labs/free-threaded-compatibility#2]
NumPy's tracking issue: https://github.com/numpy/numpy/issues/26157

Current status: fixes are going into the `main` branch of the numpy repo; the `maintenance/2.0.x` branch isn't compatible

**Comment by rgommers** [comment]
Status update:
- Test suite passing (disclaimer: with some remaining thread-safety issues not covered by tests),
- Nightly Linux wheels available at https://anaconda.org/scientific-python-nightly-wheels/numpy.

**Comment by ngoldbaum** [comment]
Nightly mac wheels are available as well now.

**Comment by ngoldbaum** [comment]
NumPy now sets the `Py_MOD_GIL_NOT_USED` slot in all its C and Cython extensions, so if you've been waiting on NumPy to start worrying about the GIL being enabled at runtime you can start working on that in your libraries.

**Comment by ngoldbaum** [comment]
It's now possible to define f2py extensions that are marked with `Py_MOD_GIL_NOT_USED`. See https://github.com/numpy/numpy/pull/26981. 

It's our understanding that f2py itself is thread safe. In the future we may add a way to add locking for f2py code that is known not to be thread safe, but for now the only way to ensure thread safety for such code is to add locking yourself or declare that the GIL is required. If you want more f2py options related to thread safety or run into thread safety issues in f2py itself please let us know by filing issues on the NumPy repo.

**Comment by ngoldbaum** [comment]
2.1.0 rc1 was released, so now there are free-threaded wheel builds on pypi. They're installable by passing `--pre` to pip.

**Comment by lysnikolaou** [comment]
Closing in favor of the upstream issue.
